presto-parquet/src/main/java/io/prestosql/parquet/reader/ParquetReader.java

```diff
@@ -206,7 +221,18 @@ private ColumnChunk readPrimitive(PrimitiveField field)
             ParquetColumnChunk columnChunk = new ParquetColumnChunk(descriptor, buffer, 0);
             columnReader.setPageReader(columnChunk.readAllPages());
         }
-        return columnReader.readPrimitive(field);
+        ColumnChunk columnChunk = columnReader.readPrimitive(field);
+
+        // update max size per primitive column chunk
+        long bytesPerCell = columnChunk.getBlock().getSizeInBytes() / batchSize;
+        if (maxBytesPerCell[fieldId] < bytesPerCell) {
+            // update batch size
+            maxBatchSize = toIntExact(min(maxBatchSize, max(1, maxReadBlockBytes / bytesPerCell)));
```

Comment:
Won't this always be ignored because `maxCombinedBytesPerRow` is strictly greater than `bytesPerCell`?